src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/AtomicReshaper.java

```diff
@@ -0,0 +1,166 @@
+import lykrast.prodigytech.common.util.Config;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RegistryDescription
+public class AtomicReshaper extends VirtualizedRegistry<AtomicReshaperManager.AtomicReshaperRecipe> {
+    @RecipeBuilderDescription(example = {
+        @Example(".input(item('minecraft:gold_ingot')).output(item('minecraft:emerald_block')).primordium(10).time(50)"),
+        @Example(".input(item('minecraft:gold_block')).addOutput(item('minecraft:diamond_block'), 10).addOutput(item('minecraft:carrot'), 3).primordium(7)")
+    })
+    public AtomicReshaper.RecipeBuilder recipeBuilder() {
+        return new AtomicReshaper.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        removeScripted().forEach(recipe -> {
+            if (recipe.isOreRecipe())
+                AtomicReshaperManager.INSTANCE.removeOreRecipe(recipe.getOreInput());
+            else
+                AtomicReshaperManager.INSTANCE.removeRecipe(recipe.getInput());
+        });
+        restoreFromBackup().forEach(AtomicReshaperManager.INSTANCE::addRecipe);
+    }
+
+    @MethodDescription(example = @Example("ore('paper')"))
+    public boolean removeByInput(IIngredient input) {
+        if (input instanceof OreDictIngredient) {
+            AtomicReshaperManager.AtomicReshaperRecipe recipe = AtomicReshaperManager.INSTANCE.removeOreRecipe(((OreDictIngredient) input).getOreDict());
+            if (recipe == null)
+                return false;
+            addBackup(recipe);
+            return true;
+        } else {
+            boolean success = false;
+            for (ItemStack it : input.getMatchingStacks()) {
+                AtomicReshaperManager.AtomicReshaperRecipe recipe = AtomicReshaperManager.INSTANCE.removeRecipe(it);
+                if (recipe != null) {
+                    success = true;
+                    addBackup(recipe);
+                }
+            }
+            return success;
+        }
+    }
+
+    public void add(AtomicReshaperManager.AtomicReshaperRecipe recipe) {
+        AtomicReshaperManager.INSTANCE.addRecipe(recipe);
+        addScripted(recipe);
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        // NOTE: can't be rolled back due to ItemMap<T> being protected
+        AtomicReshaperManager.INSTANCE.removeAll();
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "output", valid = @Comp("1"))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<AtomicReshaperManager.AtomicReshaperRecipe> {
+
+        @Property(valid = @Comp(value = "1", type = Comp.Type.GTE))
+        private int time = -1;
+
+        @Property(valid = @Comp(value = "1", type = Comp.Type.GTE))
+        private int primordium;
+
+        @Property(valid = @Comp("1"))
+        private List<Object> weightedOutputs;

```

Comment:
you should move the `new ArrayList<>();` up to here

Reply:
Yea this was just me testing problems with that bizarre crash